azurerm/resource_arm_automation_dsc_configuration.go

```diff
@@ -0,0 +1,141 @@
+	"github.com/hashicorp/terraform/helper/schema"
+	"github.com/terraform-providers/terraform-provider-azurerm/azurerm/utils"
+)
+
+func resourceArmAutomationDscConfiguration() *schema.Resource {
+	return &schema.Resource{
+		Create: resourceArmAutomationDscConfigurationCreateUpdate,
+		Read:   resourceArmAutomationDscConfigurationRead,
+		Update: resourceArmAutomationDscConfigurationCreateUpdate,
+		Delete: resourceArmAutomationDscConfigurationDelete,
+
+		Importer: &schema.ResourceImporter{
+			State: schema.ImportStatePassthrough,
+		},
+
+		Schema: map[string]*schema.Schema{
+			"name": {
+				Type:     schema.TypeString,
+				Required: true,
+				ForceNew: true,
+			},
+
+			"account_name": {
+				Type:     schema.TypeString,
+				Required: true,
+				ForceNew: true,
+			},
+
+			"resource_group_name": resourceGroupNameSchema(),
+
+			"location": locationSchema(),
+
+			"content": {
+				Type:     schema.TypeString,
+				Required: true,
+			},
+		},
+	}
+}
+
+func resourceArmAutomationDscConfigurationCreateUpdate(d *schema.ResourceData, meta interface{}) error {
+	client := meta.(*ArmClient).automationDscConfigurationClient
+	ctx := meta.(*ArmClient).StopContext
+
+	log.Printf("[INFO] preparing arguments for AzureRM Automation Dsc Configuration creation.")
+
+	name := d.Get("name").(string)
+	resGroup := d.Get("resource_group_name").(string)
+	accName := d.Get("account_name").(string)
+	content := d.Get("content").(string)
+	location := azureRMNormalizeLocation(d.Get("location").(string))
+
+	parameters := automation.DscConfigurationCreateOrUpdateParameters{
+		DscConfigurationCreateOrUpdateProperties: &automation.DscConfigurationCreateOrUpdateProperties{
+			Source: &automation.ContentSource{
+				Type:  automation.EmbeddedContent,
+				Value: &content,
+			},
+		},
+		Location: &location,
+		Name:     &name,
+	}
+
+	_, err := client.CreateOrUpdate(ctx, resGroup, accName, name, parameters)
+	if err != nil {
+		return err
+	}
+
+	read, err := client.Get(ctx, resGroup, accName, name)
+	if err != nil {
+		return err
+	}
+
+	if read.ID == nil {
+		return fmt.Errorf("Cannot read Automation Dsc Configuration '%s' (resource group %s) ID", name, resGroup)
+	}
+
+	d.SetId(*read.ID)
+
+	return resourceArmAutomationDscConfigurationRead(d, meta)
+}
+
+func resourceArmAutomationDscConfigurationRead(d *schema.ResourceData, meta interface{}) error {
+	client := meta.(*ArmClient).automationDscConfigurationClient
+	ctx := meta.(*ArmClient).StopContext
+
+	id, err := parseAzureResourceID(d.Id())
+	if err != nil {
+		return err
+	}
+	resGroup := id.ResourceGroup
+	accName := id.Path["automationAccounts"]
+	name := id.Path["configurations"]
+
+	resp, err := client.Get(ctx, resGroup, accName, name)
+	if err != nil {
+		if utils.ResponseWasNotFound(resp.Response) {
+			d.SetId("")
+			return nil
+		}
+
+		return fmt.Errorf("Error making Read request on AzureRM Automation Dsc Configuration '%s': %+v", name, err)
+	}
+
```

Comment:
Is there a reason you are not fetching the content? It does appear to be part of the data structure. If there is a reason a comment explaining why here would be great.

Reply:
Also is there a reason location not being read back into state?

Reply:
Content is not part of the data structure, instead it needs to be fetched by GetContent function. I have an issue with the Azure Go SDK so I won't be able to fix this right now. Reported an [issue](https://github.com/Azure/azure-sdk-for-go/issues/2486)

Reply:
Looks like that issue has been resolved